pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -939,6 +942,31 @@ type NomadSDConfig struct {
 	// +optional
 	EnableHTTP2 *bool `json:"enableHTTP2,omitempty"`
 }
+
+// OVHCloudSDConfig configurations allow retrieving scrape targets from OVHcloud's dedicated servers and VPS using their API.
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ovhcloud_sd_config
+// +k8s:openapi-gen=true
+type OVHCloudSDConfig struct {
+	// Access key to use. https://api.ovh.com.
+	// +required
+	ApplicationKey string `json:"applicationKey"`
+	// +required
+	ApplicationSecret *corev1.SecretKeySelector `json:"applicationSecret"`
```

Comment:
Since this is a required field, can we just use the value here instead of a pointer?
```suggestion
	ApplicationSecret corev1.SecretKeySelector `json:"applicationSecret"`
```

Reply:
Got it.